src/common/metrics/TemporalMetric.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Keeps track of a metric temporally in order to produce it's rate over an interval of time.
+ *
+ * Counts are stored in a specified number of slots each representing an equal amount of time.
+ * Counts are accumilated in a slot until the interval of time has passed, at which point counts
+ * start accumilating in a new slot. There are a limited number of slots and each time a new slot is
+ * used, the oldest is discarded.
+ *
+ * This class currenly only supports uint counts.
+ */
+
+#include <SFML/System.hpp>
+#include <boost/circular_buffer.hpp>
+
+#ifndef TemporalMetric_hpp
+#define TemporalMetric_hpp
+
+class TemporalMetric {
```

Comment:
can you add some tests for this? It would suck if we got it wrong

Reply:
I was going to but then I realized how difficult it would be without mocking time.
Any ideas?